src/index.cpp

```diff
@@ -1255,6 +1298,8 @@ void Index<T, TagT, LabelT>::occlude_list(const uint32_t location, std::vector<N
                 {
                     uint32_t a = iter->id;
                     uint32_t b = iter2->id;
+                    if (_pts_to_labels.size() < b || _pts_to_labels.size() < a)
```

Comment:
under what scenario will this occur? i guess unfiltered index will have size == 0. anything else?

Reply:
This was added a sanitary check, In case of Filtered Streaming Index pts_to_labels will have size num_points + num_frozen_points. In case where num_frozen_points > unique labels.  and the candidate has a frozen_point without filter. It results in program crash.

Reply:
can you elaborate bit more? if we continue, we wont update the occlude factor etc. right? wont that be incorrect behavior, especially for non-filtered index?

Reply:
For example, it eems like the for loop below insists that a has all the labels that b has. So we can allow empty labels for b right?